src/include/txn_inline.h

```diff
@@ -856,6 +858,16 @@ __wt_txn_read_upd_list(
         if (type == WT_UPDATE_RESERVE)
             continue;
 
+        /*
+         * Save the restored update to use it as base value update in case if we need to reach
+         * history store instead of on-disk value.
+         */
+        if (restored_updp != NULL && F_ISSET(upd, WT_UPDATE_RESTORED_FROM_HS) &&
```

Comment:
We restored this update from hs and then use it as a base value?

Reply:
Yes. If there is a restored update in the update list that doesn't satisfy the read request, so it must be the base for all the history store searches than the on-disk update.